atspi/examples/doc-info-tree.rs

```diff
@@ -0,0 +1,86 @@
+//! This example watches for document load events and prints a tree of accessible objects
+//! descending from the document root. It displays some generally useful info like
+//! the name, role, and coordinates of each accessible object.
+//!
+//! ```sh
+//! cargo run --example doc-info-tree
+//! ```
+//! Authors:
+//!    Colton Loftus,
+//!    Luuk van der Duim,
+//!    Tait Hoyem
+
+use atspi::connection::set_session_accessibility;
+use atspi::proxy::accessible::{AccessibleProxy, ObjectRefExt};
+use atspi::proxy::proxy_ext::ProxyExt;
+use atspi::{DocumentEvents, Event, State};
+use std::error::Error;
+use tokio_stream::StreamExt;
+use zbus::Connection;
+
+// A type alias for the future returned by recursive_print_children
+type BoxedFuture<'a> =
+	std::pin::Pin<Box<dyn std::future::Future<Output = Result<(), Box<dyn Error>>> + Send + 'a>>;
+
+/// recursively print the children of an accessible object
+/// along with some useful information like their name and
+/// coordinates
+fn recursive_print_children<'a>(
+	proxy: &'a AccessibleProxy<'a>,
+	conn: &'a Connection,
+	indent: usize,
+) -> BoxedFuture<'a> {
+	Box::pin(async move {
+		for child in proxy.get_children().await?.iter() {
+			let child_proxy = child.as_accessible_proxy(conn).await?;
+			let component = child_proxy.proxies().await?.component().await?;
+			let extents = component.get_extents(atspi::CoordType::Screen).await?;
+			let name = child_proxy.name().await?;
+			let role = child_proxy.get_role_name().await?;
+			if child_proxy.get_state().await?.contains(State::Focusable) {
+				println!(
+					"{}child, role:{}, name:{}, extents:{:?}",
+					" ".repeat(indent),
```

Comment:
Usually Clippy will tell you to inline the argument in format and printiln macros.